scripts/serverless-setup.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Create and commit .sam-params-<env> file as SAM deploy parameters
+# The cert ARN and hostedZoneID are not sensitive but requires list all permissions in IAM
+# therefore are created here instead of granting the CI user those permission to fetch on the fly
+# and they should not change often
+
+if [[ "$ENVIRONMENT" == "stage" ]]; then
+  BACKEND_DOMAIN_PREFIX=$stagingBackendSubdomain
+  DOMAIN=$stagingHostRoot
+  FILE_NAME=".sam-params-stage"
+elif [[ "$ENVIRONMENT" == "prod" ]]; then
+  BACKEND_DOMAIN_PREFIX=$productionBackendSubdomain
+  DOMAIN=$productionHostRoot
+  FILE_NAME=".sam-params-prod"
+fi
+
+HOSTED_ZONE_ID=$(aws route53 list-hosted-zones  --query "HostedZones[?Name == '${DOMAIN}.'].Id"  | jq -r '.[0]' | sed 's/\/hostedzone\///')
+GATEWAY_CERTIFICATE_ARN=$(aws acm list-certificates --region ${REGION} | jq -r " .CertificateSummaryList [] | select( .DomainName == \"${BACKEND_DOMAIN_PREFIX}${DOMAIN}\") | .CertificateArn")
+
+cd $PROJECT_DIR
+if [ -f "$FILE_NAME" ]; then
+  echo "$FILE_NAME already exist, skipping deploy params generation."
+else
+  cat <<-EOF > $FILE_NAME
+Environment=$ENVIRONMENT
+HostedZoneId=$HOSTED_ZONE_ID
+GatewayCertificateArn=$GATEWAY_CERTIFICATE_ARN
+
+EOF
+  # Commit the parameters needed for deployment
+  git add $FILE_NAME
+  git commit -m 'Add Hostedzone and Cert arn' && git push
```

Comment:
This feels weird.. Maybe we can think of a different way.

Reply:
we could also add them to SSM/secret manager and reference them like secrets

Reply:
Yeah, parameter store might be a good spot. In regular usage the only place that will use this file is the deploy pipeline, right?

Reply:
yeah it should only be needed for deployment, looks like `build` / `start-api` locally doesn't need those parameters.